<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from surgical_tool.urdf.xacro       | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="surgical_robot_control">
  <!-- <xacro:include filename="my_materials.xacro"/> -->
  <!-- <xacro:include filename="./colors.urdf.xacro" /> -->
  <link name="base">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1"/>
      <mass value="0.1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1"/>
      <geometry>
        <box size="0.05 0.05 0.2"/>
      </geometry>
      <material name="base">
        <color rgba="1.0 0.0 0.0 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1"/>
      <geometry>
        <box size="0.05 0.05 0.2"/>
      </geometry>
    </collision>
  </link>
  <joint name="base_actuator" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    <parent link="base"/>
    <child link="actuator"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="30" lower="-3.14" upper="3.14" velocity="1"/>
  </joint>
  <link name="actuator">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mesh filename="package://surgical_robot_control/meshes/ActuatorModule_Blender_zero.dae"/>
      <mass value="0.1"/>
      <inertia ixx="4.166666666666667e-05" ixy="0.0" ixz="0.0" iyy="4.166666666666667e-05" iyz="0.0" izz="0.0006666666666666668"/>
    </inertial>
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/ActuatorModule_Blender_zero.dae"/>
      </geometry>
      <!-- <material name="actuator">
                <color rgba="1.0 1.0 0.0 0.4"/>
            </material> -->
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/ActuatorModule_Blender_zero.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="actuator_shaft" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="0.187 0.0 0.0"/>
    <parent link="actuator"/>
    <child link="shaft"/>
  </joint>
  <link name="shaft">
    <inertial>
      <origin rpy="0.0 2.0 0.0" xyz="0.25 0.0 0.0"/>
      <mass value="0.1"/>
      <inertia ixx="4.166666666666667e-05" ixy="0.0" ixz="0.0" iyy="4.166666666666667e-05" iyz="0.0" izz="0.0006666666666666668"/>
    </inertial>
    <visual>
      <origin rpy="0.0 1.5708 0.0" xyz="0.1 0.0 0.0"/>
      <geometry>
        <cylinder length="0.2" radius="0.0015"/>
      </geometry>
      <material name="shaft">
        <color rgba="1.0 1.0 0.0 0.3"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.1 0.0 0.0"/>
      <geometry>
        <cylinder length="0.2" radius="0.0015"/>
      </geometry>
    </collision>
  </link>
  <joint name="tilt_1" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.2 0.0 0.0"/>
    <parent link="shaft"/>
    <child link="seg_tilt_1"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="30.0" lower="-0.7854" upper="0.7854" velocity="3.141"/>
  </joint>
  <link name="seg_tilt_1">
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/continuum_segment_Blender_zero_lightgreen.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>
  <joint name="pan_1" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0018 0.0 0.0"/>
    <parent link="seg_tilt_1"/>
    <child link="seg_pan_1"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="30.0" lower="-0.7854" upper="0.7854" velocity="3.141"/>
  </joint>
  <link name="seg_pan_1">
    <visual>
      <origin rpy="1.5708 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/continuum_segment_Blender_zero_violet.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>
  <joint name="tilt_2" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0018 0.0 0.0"/>
    <parent link="seg_pan_1"/>
    <child link="seg_tilt_2"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="30.0" lower="-0.7854" upper="0.7854" velocity="3.141"/>
  </joint>
  <link name="seg_tilt_2">
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/continuum_segment_Blender_zero_lightgreen.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>
  <joint name="pan_2" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0018 0.0 0.0"/>
    <parent link="seg_tilt_2"/>
    <child link="seg_pan_2"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="30.0" lower="-0.7854" upper="0.7854" velocity="3.141"/>
  </joint>
  <link name="seg_pan_2">
    <visual>
      <origin rpy="1.5708 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/continuum_segment_Blender_zero_violet.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>
  <joint name="tilt_3" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0018 0.0 0.0"/>
    <parent link="seg_pan_2"/>
    <child link="seg_tilt_3"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="30.0" lower="-0.7854" upper="0.7854" velocity="3.141"/>
  </joint>
  <link name="seg_tilt_3">
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/continuum_segment_Blender_zero_lightgreen.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>
  <joint name="pan_3" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0018 0.0 0.0"/>
    <parent link="seg_tilt_3"/>
    <child link="seg_pan_3"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="30.0" lower="-0.7854" upper="0.7854" velocity="3.141"/>
  </joint>
  <link name="seg_pan_3">
    <visual>
      <origin rpy="1.5708 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/continuum_segment_Blender_zero_violet.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>
  <joint name="tilt_4" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0018 0.0 0.0"/>
    <parent link="seg_pan_3"/>
    <child link="seg_tilt_4"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="30.0" lower="-0.7854" upper="0.7854" velocity="3.141"/>
  </joint>
  <link name="seg_tilt_4">
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/continuum_segment_Blender_zero_lightgreen.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>
  <joint name="pan_4" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0018 0.0 0.0"/>
    <parent link="seg_tilt_4"/>
    <child link="seg_pan_4"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="30.0" lower="-0.7854" upper="0.7854" velocity="3.141"/>
  </joint>
  <link name="seg_pan_4">
    <visual>
      <origin rpy="1.5708 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/continuum_segment_Blender_zero_violet.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>
  <joint name="tilt_5" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0018 0.0 0.0"/>
    <parent link="seg_pan_4"/>
    <child link="seg_tilt_5"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="30.0" lower="-0.7854" upper="0.7854" velocity="3.141"/>
  </joint>
  <link name="seg_tilt_5">
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/continuum_segment_Blender_zero_lightgreen.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>
  <joint name="pan_5" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0018 0.0 0.0"/>
    <parent link="seg_tilt_5"/>
    <child link="seg_pan_5"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="30.0" lower="-0.7854" upper="0.7854" velocity="3.141"/>
  </joint>
  <link name="seg_pan_5">
    <visual>
      <origin rpy="1.5708 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://surgical_robot_control/meshes/continuum_segment_Blender_zero_violet.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>
</robot>
